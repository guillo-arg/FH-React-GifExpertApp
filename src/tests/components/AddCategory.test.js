import { AddCategory } from "../../components/AddCategory";
import {shallow} from 'enzyme';

describe('pruebas sobre <AddCategory />', () => {
    const setCategories =jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories}/>);

    });

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe cambiar la caja de texto', () => {
      const input = wrapper.find('input');
      const value = 'Hola mundo';
      input.simulate('change', {
          target : {
              value: value
          }
      });
    });

    
    
    test('No debe postear la informaciÃ³n', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        
        expect(setCategories).not.toHaveBeenCalled();
    });
    
    test('Debe llamar el setCategories y limpiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {
            target : {
                value: value
            }
        });

        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).toHaveBeenCalled();
        expect(input.prop('value')).toBe('');


    });
    
    
});
