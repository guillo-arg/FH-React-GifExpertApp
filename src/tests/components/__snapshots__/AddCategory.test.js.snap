// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre <AddCategory /> Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
